src/file-utils/edfi_lms_file_utils/file_reader.py

```diff
@@ -18,17 +18,42 @@ def _default() -> pd.DataFrame:
     return pd.DataFrame()
 
 
-def _read_csv(file: str, nrows: Union[int, None] = None) -> pd.DataFrame:
+def _read_csv(
+    file: str,
+    nrows: Optional[int] = None,
+    data_types: Dict[str, str] = dict(),
+    date_cols: List[str] = list(),
+) -> pd.DataFrame:
     logger.info(f"Reading file: {file}")
     if file:
+
+        dates = [
+            "SourceCreateDate",
+            "SourceLastModifiedDate",
+            "CreateDate",
+            "LastModifiedDate",
+            *date_cols,
+        ]
+
+        dtype = {
+            **data_types,
+            "SourceSystemIdentifier": "string",
+            "SourceSystem": "string"
+        }
+
         return pd.read_csv(
-            file, engine="c", parse_dates=True, infer_datetime_format=True, nrows=nrows
+            file,
+            engine="c",
+            parse_dates=dates,
+            infer_datetime_format=True,
+            nrows=nrows,
+            dtype=dtype,
```

Comment:
Here are the changes around data types. Note that the function signature now accepts two optional parameters, for defining columns that should be treated as dates, and for other data types (which will typically be strings, but could be int64 or float). Note that there are some common columns that exist in all cases (I think), so there is no need to retype `SourceSystemIdentifier` each time, for example.